src/olympia/files/helpers.py

```diff
@@ -60,6 +64,32 @@ def file_tree(files, selected):
     return jinja2.Markup('\n'.join(output))
 
 
+def extract_file(viewer, **kw):
+    # This message is for end users so they'll see a nice error.
+    msg = Message('file-viewer:%s' % viewer)
+    msg.delete()
+    task_log.debug('Unzipping %s for file viewer.' % viewer)
+
+    try:
+        locked = viewer.extract()
+
+        if locked:
+            info_msg = _(
+                'File viewer is locked, extraction for %s could be '
+                'in progress. Please try again in approximately 5 minutes.'
+                % viewer)
+            msg.save(info_msg)
+    except Exception, err:
+        if settings.DEBUG:
+            msg.save(_('There was an error accessing file %s. %s.') %
```

Comment:
Ahhh, great catch! Again! The joys of copying things and not noticing details like that. it'd be nice to switch between details and not based on permission but for now I'll change that string and just append the error.